### PR TITLE
external storage: disable s3 100 continue

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/38089

Problem Summary:

### What is changed and how it works?
Add a disable s3 100 continue hotfix.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by D3Hunter — APPROVED** [review]

File: br/pkg/storage/s3.go [hunk 1]
```diff
@@ -276,7 +276,8 @@ func newS3Storage(backend *backuppb.S3, opts *ExternalStorageOptions) (obj *S3St
 	qs := *backend
 	awsConfig := aws.NewConfig().
 		WithS3ForcePathStyle(qs.ForcePathStyle).
-		WithRegion(qs.Region)
+		WithRegion(qs.Region).
+		WithS3Disable100Continue(true)
```
Comment (3pointer) [inline br/pkg/storage/s3.go:280]: will we merge this code to master?

Comment (lichunzhu) [reply br/pkg/storage/s3.go:280]: No. If we want to merge this to master, I think we need to add an argument `S3Disable100Continue` in `qs` and auto-adjust this argument through `s3.point` or `s3.provider`. This PR is used only for verification.

Comment (3pointer) [reply br/pkg/storage/s3.go:280]: but the hotfix code should merge into master according to release team

Comment (lichunzhu) [reply br/pkg/storage/s3.go:280]: Never heared of that before.... If we can't make sure this is the cause of this problem, how can we directly merge this into master?

**Review by 3pointer — APPROVED** [review]

**Comment by 3pointer** [comment]
lgtm(but we should change the code if we merge it into master)

**Pull request merged**